internal/services/pubkey_service.go

```diff
@@ -15,25 +19,67 @@ func CreatePubkey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	pubkeyDao, err := dao.GetPubkeyDao(r.Context())
-	if err != nil {
-		renderError(w, r, payloads.NewInitializeDAOError(r.Context(), "pubkey DAO", err))
-		return
-	}
+	// TODO: extract this into job queue
+	err := dao.WithTransaction(r.Context(), db.DB, func(tx dao.Transaction) error {
+		pkDao, err := dao.GetPubkeyDao(r.Context(), tx)
+		if err != nil {
+			return payloads.NewInitializeDAOError(r.Context(), "pubkey DAO", err)
+		}
```

Comment:
This is how you do transactions with this proposal - you simply call a function and pass `tx` into Dao "getter", that's all. All the rest is the same.